Add additional info to dictionary entries and use standard arg parsing

mymro opened the pull request:
This pull request resolves #14 and uses the standard arg parser.

mymro committed:
Add support for info in dictionary entries


Cleanup


Update README


Fix bug in JMnedict parsing


Update Makefile


Fix Bug in minifying index


Fix Bug in minifying index


Fix Bug in minifying index


Update README


Rename

mymro committed:
Add additional Info to entries

mymro committed:
Make it clearer that the combined dictionary is missing some features

mymro commented:
@jrfonseca When will you merge the commits?

jrfonseca commented on README.md:
```diff
@@ -20,9 +20,6 @@ then processed with ImageMagick's
 -->
-
-![Inflection lookup screenshot](screenshots/infl.png)
-
```
I'd like to keep an example of inflection.

jrfonseca commented on dictionary.py:
```diff
@@ -237,14 +241,14 @@ def write_index(entries, dictionary_name, title, stream, respect_re_restr=True,
             for sentence in entry.sentences:
                 stream.write(' <div class="sen">\n')
-                stream.write('  <span>' + sentence.japanese + '</span>\n')
+                stream.write(f"  <span>{sentence.japanese}</span>\n")
```
f-strings is great, but for future reference, I'd prefer if we standardized on single quotes `'` strings, as this makes it easier to emit `"` in the generated XML/HTML

jrfonseca commented on dictionary.py:
```diff
@@ -270,29 +274,28 @@ def write_index(entries, dictionary_name, title, stream, respect_re_restr=True,
     stream.write('    <dc-metadata xmlns:dc="http://purl.org/metadata/dublin_core">\n')
-    stream.write('      <dc:Identifier id="uid">%s</dc:Identifier>\n' %(hex(hash(title)).split('x')[1]))
-    stream.write('      <dc:Title><h2>%s</h2></dc:Title>\n' %title)
+    stream.write(f"      <dc:Identifier id=\"uid\">{hex(hash(title)).split('x')[1]}</dc:Identifier>\n")
```
For example, here, using `f'...'` would avoid escaping the single quote.

jrfonseca commented on jmdict.py:
```diff
@@ -404,85 +410,89 @@ def parse_translation(self):
+
+    arg_parser = argparse.ArgumentParser(description='Build the JMdict or JMnedict for Kindle', formatter_class=argparse.RawTextHelpFormatter)
+    arg_parser.add_argument('-s', '--sentences'
+        ,action='store'
```
I don't want to nitpick, but commas like this look awkward to me.

I'm starting to think it might be better to start using some sort of automated Python code formatting tool to ensure code is consistent.

mymro replied:
@jrfonseca How would you prefer them?

jrfonseca replied:
Trailing commas.  https://www.python.org/dev/peps/pep-0008/#when-to-use-trailing-commas

Honestly, I think the easiest thing to do is to use an automated tool to format like https://www.python.org/dev/peps/pep-0008/  .  I know there are a few, and I even recall to use one a couple of years ago, but I forgot which.

jrfonseca reviewed (commented):
Otherwise looks good  AFACT.

Sorry for the delay.  I've been really busy with some personal stuff, and will continue to be, until past Easter.

mymro commented:
@jrfonseca I will probably update this commit over the weekend

mymro committed:
Add auto format with Black to Github Action


Fix auto format in action


Fix auto format in action


Fix auto format in action


Fix auto format in action


a


a


Change Actions for pull_request_target


a


a


a

mymro committed:
Apply Format with Black

mymro committed:
Add Inflection Image back

mymro committed:
Merge branch 'master' into master

mymro committed:
update formatting

mymro committed:
Fix merging error

mymro committed:
Make README nicer


a


a

mymro commented:
@jrfonseca The code will now be checked for being correctly formatted with black in each push or pull request. In the README there is a section that explains how to automatically let black format the source files.

I have also included the screenshot with an inflection and put the images on two lines in order for them not to take up so much space.

As soon as this pull request is merged, I will check the whole repository for compliance with black and then push eventual changes.

mymro commented:
@jrfonseca Could you please merge the pull request.

jrfonseca commented:
Thanks.
